src/base/memory_alloc.h

```diff
@@ -155,19 +101,34 @@ extern void db_scramble (void *region, int size);
           os_free((ptr)); \
           (ptr) = NULL; \
         } while (0)
-#endif /* NDEBUG */
 
-extern int ansisql_strcmp (const char *s, const char *t);
-extern int ansisql_strcasecmp (const char *s, const char *t);
+#define db_ws_free_and_init(ptr) \
+        do { \
+          db_ws_free((ptr)); \
+          (ptr) = NULL; \
+        } while (0)
+#endif /* NDEBUG */
 
 #if !defined (SERVER_MODE)
 
 extern HL_HEAPID private_heap_id;
+extern HL_HEAPID ws_heap_id;
 
 #define os_malloc(size) (malloc (size))
 #define os_free(ptr) (free (ptr))
 #define os_realloc(ptr, size) (realloc ((ptr), (size)))
 
+/* allocation APIs for workspace */
+#define db_ws_alloc(size) \
+        db_private_alloc(NULL, size)
```

Comment:
It is trivial.

Isn't our coding convention to put a space after the function name?
It seems to be mixed.
```
#define db_ws_alloc(size) \
        db_private_alloc (NULL, size)
```

Reply:
In the macro alias, I found that space is not used
```
#if !defined(NDEBUG)
#define db_private_alloc(thrd, size) \
        db_private_alloc_debug(thrd, size, true, __FILE__, __LINE__)
#define db_private_free(thrd, ptr) \
        db_private_free_debug(thrd, ptr, true, __FILE__, __LINE__)
#define db_private_realloc(thrd, ptr, size) \
        db_private_realloc_debug(thrd, ptr, size, true, __FILE__, __LINE__)
...
```